aten/src/ATen/native/ForeachOpsKernels.cpp

```diff
@@ -42,4 +98,61 @@ void foreach_tensor_add_list_kernel_slow_(TensorList tensors1, TensorList tensor
   }
 }
 
+std::vector<Tensor> foreach_tensor_sub_list_kernel_slow(TensorList tensors1, TensorList tensors2) {
+  verify_list(tensors1, tensors2);
+
+  std::vector<Tensor> result;
+  for (int i = 0; i < tensors1.size(); i++) {
+    result.emplace_back(tensors1[i].sub(tensors2[i]));
+  }
+
+  return result;
+}
+
+void foreach_tensor_sub_list_kernel_slow_(TensorList tensors1, TensorList tensors2) {
+  verify_list(tensors1, tensors2);
+
+  for (int i = 0; i < tensors1.size(); i++) {
+    tensors1[i].sub_(tensors2[i]);
+  }
+}
+
+std::vector<Tensor> foreach_tensor_mul_list_kernel_slow(TensorList tensors1, TensorList tensors2) {
+  verify_list(tensors1, tensors2);
+
+  std::vector<Tensor> result;
+  for (int i = 0; i < tensors1.size(); i++) {
+    result.emplace_back(tensors1[i].mul(tensors2[i]));
+  }
+
+  return result;
+}
+
+void foreach_tensor_mul_list_kernel_slow_(TensorList tensors1, TensorList tensors2) {
+  verify_list(tensors1, tensors2);
+
+  for (int i = 0; i < tensors1.size(); i++) {
+    tensors1[i].mul_(tensors2[i]);
+  }
+}
+
+std::vector<Tensor> foreach_tensor_div_list_kernel_slow(TensorList tensors1, TensorList tensors2) {
```

Comment:
Would prefer some templates here for DRY. e.g.,

```
template <Tensor (*Op)(const Tensor&, const Tensor&)>
void foreach_tensor_op_list_kernel_slow_(TensorList tensors1, TensorList tensors2) {
    verify_list(tensors1, tensors2)
    for (...) {
        result.emplace_back(Op(tensors1[i], tensors2[i]);
    }
}
```

Reply:
Actually, I see the conundrum. native_functions.yaml wants the name of the function and probably can't take a template.

Reply:
We could use function pointers that point to an instantiated version of the above template. But all of that seems like overkill for now. Why not use a simple macro?